.github/workflows/docs.yml

```diff
@@ -81,20 +81,14 @@ jobs:
           cd build/html
           touch .nojekyll
           git add -A --force .
-          git commit -m "Automated documentation rebuild"
+          git commit -m "Automated documentation rebuild" || true  # allow this to fail if no changes were made
```

Comment:
I see that this will resolve the issue of if there are no changes to commit, but what if there is an issue with the `git commit` step that's due to some separate issue? I think this is how we saw there was a problem with the documentation builder in the first place right?
Maybe this can be modified to something like `git commit -m "Automated documentation rebuild" || <some code that will return true if there are no changes to commit> `
Or before `git add -A --force .`, can check to see if there are any files to commit with something like `git diff --exit-code` or `git diff --quiet`.

Reply:
This is a good point - let me add this now.